pkg/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -2747,6 +2770,71 @@ func TestProcessNetworkPolicy(t *testing.T) {
 	}
 }
 
+func TestProcessNetworkPolicyLogging(t *testing.T) {
+	tests := []struct {
+		name                    string
+		inputPolicy             *networkingv1.NetworkPolicy
+		expectedPolicy          *antreatypes.NetworkPolicy
+		expectedAppliedToGroups int
+		expectedAddressGroups   int
+	}{
+		{
+			name: "default-allow-ingress",
+			inputPolicy: &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{Namespace: "nsA", Name: "npA", UID: "uidA"},
+				Spec: networkingv1.NetworkPolicySpec{
+					PodSelector: metav1.LabelSelector{},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					Ingress:     []networkingv1.NetworkPolicyIngressRule{{}},
+				},
+			},
+			expectedPolicy: &antreatypes.NetworkPolicy{
+				UID:  "uidA",
+				Name: "uidA",
+				SourceRef: &controlplane.NetworkPolicyReference{
+					Type:      controlplane.K8sNetworkPolicy,
+					Namespace: "nsA",
+					Name:      "npA",
+					UID:       "uidA",
+				},
+				Rules: []controlplane.NetworkPolicyRule{{
+					Direction:     controlplane.DirectionIn,
+					From:          matchAllPeer,
+					Services:      nil,
+					Priority:      defaultRulePriority,
+					Action:        &defaultAction,
+					EnableLogging: true,
+				}},
+				AppliedToGroups: []string{getNormalizedUID(antreatypes.NewGroupSelector("nsA", &metav1.LabelSelector{}, nil, nil, nil).NormalizedName)},
+			},
+			expectedAppliedToGroups: 1,
+			expectedAddressGroups:   0,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, c := newController()
+			// Replace with custom lister that returns Namespace with logging Annotation.
+			testNamespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace:   corev1.NamespaceDefault,
+					Name:        "nsA",
+					Annotations: map[string]string{"policy.antrea.io/enable-np-logging": "true"},
+				},
+			}
```

Comment:
```suggestion
			_, c := newController()
			testNamespace := &corev1.Namespace{
				ObjectMeta: metav1.ObjectMeta{
					Namespace:   corev1.NamespaceDefault,
					Name:        "nsA",
					Annotations: map[string]string{"policy.antrea.io/enable-np-logging": "true"},
				},
			}
			c.namespaceStore.Add(testNamespace)
```
then you don't have to create a new mock namespaceLister.